WordPress/Classes/Utility/DiskCache.swift

```diff
@@ -0,0 +1,273 @@
+    public init(url: URL, configuration: Configuration = .init()) {
+        self.rootURL = url
+        self.configuration = configuration
+
+        do {
+            try FileManager.default.createDirectory(at: rootURL, withIntermediateDirectories: true, attributes: nil)
+        } catch {
+            DDLogError("Failed to creates cache root directory at: \(url) with error: \(error)")
+        }
+        DispatchQueue.main.asyncAfter(deadline: .now() + configuration.initialSweepDelay) { [weak self] in
+            Task {
+                await self?.performAndScheduleNextSweep()
+            }
+        }
+    }
+
+    // MARK: Codable
+
+    public func getValue<T: Decodable>(
+        _ type: T.Type,
+        forKey key: String,
+        decoder: JSONDecoder = JSONDecoder()
+    ) async -> T? {
+        guard let data = getData(forKey: key) else { return nil }
+        return try? decoder.decode(type, from: data)
+    }
+
+    public func setValue<T: Encodable>(
+        _ value: T,
+        forKey key: String,
+        encoder: JSONEncoder = JSONEncoder()
+    ) async {
+        guard let data = try? encoder.encode(value) else { return }
+        setData(data, forKey: key)
+    }
+
+    public func removeValue(forKey key: String) {
+        removeData(forKey: key)
+    }
+
+    // MARK: Codable (Closures)
+
+    public nonisolated func getValue<T: Decodable>(
+        _ type: T.Type,
+        forKey key: String,
+        decoder: JSONDecoder = JSONDecoder(),
+        _ completion: @escaping (T?) -> Void
+    ) {
+        Task {
+            let value = await getValue(type, forKey: key, decoder: decoder)
+            completion(value)
+        }
+    }
+
+    public nonisolated func setValue<T: Encodable>(
+        _ value: T,
+        forKey key: String,
+        encoder: JSONEncoder = JSONEncoder()
+    ) {
+        Task {
+            await setValue(value, forKey: key, encoder: encoder)
+        }
+    }
+
+    // MARK: Accessing Cached Data
+
+    public func getData(forKey key: String) -> Data? {
+        guard let url = fileURL(for: key) else { return nil }
+        return try? Data(contentsOf: url)
+    }
+
+    public func setData(_ data: Data, forKey key: String) {
+        guard let url = fileURL(for: key) else { return }
+        do {
+            try data.write(to: url)
+        } catch let error as NSError {
+            guard error.code == CocoaError.fileNoSuchFile.rawValue && error.domain == CocoaError.errorDomain else { return }
+            try? FileManager.default.createDirectory(at: rootURL, withIntermediateDirectories: true, attributes: nil)
+            try? data.write(to: url) // re-create a directory and try again
+        }
+    }
+
+    public func removeData(forKey key: String) {
+        guard let url = fileURL(for: key) else { return }
+        try? FileManager.default.removeItem(at: url)
+    }
+
+    /// Removes all cached entries.
+    public func removeAll() {
+        do {
```

Comment:
Same thing here – we shouldn't just ignore IO errors

Reply:
Made it throwing.